rex/resource.py

```diff
@@ -148,9 +212,95 @@ def _check_slice(ds_slice):
 
         return ds_slice, ds_idx
 
+    def _extract_list_slice(self, ds_slice):
+        """
+        Optimize and extract list slice request along a single dimension
+
+        Parameters
+        ----------
+        ds_slice : int | slice | list | ndarray | tuple
+            What to extract from ds, each arg is for a sequential axis
+
+        Returns
+        -------
+        out : ndarray
+            Extracted array of data from ds
+        """
+        out_slices = []
+        chunks = self.chunks
+        sort_idx = []
+        list_len = None
+        if chunks:
+            for i, ax_slice in enumerate(ds_slice):
+                c = chunks[i]
+                if isinstance(ax_slice, (list, np.ndarray)):
+                    if not isinstance(ax_slice, np.ndarray):
+                        ax_slice = np.array(ax_slice)
+
+                    idx = np.argsort(ax_slice)
```

Comment:
This is kind of the crux of it right? i'm a little confused on how this all fits together including list_len.